fe/fe-core/src/main/java/org/apache/doris/nereids/stats/StatsCalculator.java

```diff
@@ -464,6 +478,32 @@ private Statistics computeTopN(TopN topN) {
         return stats.withRowCount(Math.min(stats.getRowCount(), topN.getLimit()));
     }
 
+    private Statistics computePartitionTopN(PartitionTopN partitionTopN) {
+        Statistics stats = groupExpression.childStatistics(0);
+        double rowCount = stats.getRowCount();
+        List<Expression> partitionKeys = partitionTopN.getPartitionKeys();
+        if (!partitionTopN.hasGlobalLimit() && !partitionKeys.isEmpty()) {
```

Comment:
For the limit part, I don't think it will cause some bad cases.
```
private Statistics computeLimit(Limit limit) {
        Statistics stats = groupExpression.childStatistics(0);
        return stats.withRowCount(Math.min(stats.getRowCount(), limit.getLimit()));
    }
```
Because we only get the minimal value.

Reply:
`select *  from (select * ,rank() over(partition by b order by a) as num from t ) tt where  num < 10;`
rank() column stats should be set manually, its max is 10. And hence, outer filter `num<10` selectivity is 1.

Reply:
Maybe it's hard to handle because for the `PartitionTopN` node, we do not get the column for the `rank`. So just add the TODO.